tests/torchtune/utils/test_precision.py

```diff
@@ -30,7 +30,7 @@ def test_get_dtype(self):
         """
         Tests that the correct dtype is returned based on the input string.
         """
-        dtypes = [None, torch.half] + list_dtypes()
+        dtypes = [None, torch.half] + list(PRECISION_STR_TO_DTYPE.keys())
```

Comment:
Global variables like PRECISION_STR_TO_DTYPES should not be public, which is the reason for a function like list_dtypes. This also allows us to add any logic on top of the list if we need to in the future without breaking backward compatibility.

Reply:
So (a) we are not really using `PRECISION_STR_TO_DTYPE` publicly or exposing it as a public API in any official capacity, right? We are already using it in `get_dtype` anyways, I just added one more usage to remove the `list_dtypes` API. And (b) it's similar to my comment about `transform_opt_state_dict` -- we are doing extra indirection for no real reason here. Also not sure I follow the BC-breaking argument.. like no matter what this is just a global mapping that we update with new(/deprecated) dtypes, we're not gonna change `list_dtypes` and so we break BC in the same way no matter what (unless I'm missing something obvious here)

Reply:
To the BC comment, get_list is essentially a getter to the mapping. I believe it's best practices to use a getter instead of directly exposing the mapping because you'd run into problems if you ever needed to add any dynamic logic (like filtering on hardware, etc). Either way this is a nit, so it's up to you.